### PR TITLE
feat: allow filtering of emails in http endpoint

### PR DESCRIPTION
**Pull request by madflow**
* This PR allows simple filtering in the HTTP endpoint `/email`
* Use case is - when using in CI (Cypress E2E), it can be useful to only fetch specific emails. Also #330 
* You can only find emails, which correspond exactly to the query parameters.
* Nested fields in the payload can be addressed with a (`to.name`, `headers.x-some-header`) dot syntax

`http http://localhost:1080/email\?to.name\="Johnny Utah"\&subject\="The ex-presidents are surfers"`

### CODE REVIEW

**Comment by kowalczyka** [comment]
Hello,
does this change allows filtering only for endpoint in REST API or does it also affects Web UI?

**Comment by madflow** [comment]
> Hello,
> does this change allows filtering only for endpoint in REST API or does it also affects Web UI?

Only for simple filtering in the HTTP endpoint /email.

File: test/utils.test.js [hunk 1]
```diff
@@ -14,4 +14,20 @@ describe('utils', () => {
...
   })
+
+  describe('filterEmails', () => {
+    const emails = [
+      { subject: 'Test', headers: { 'x-some-header': 1 } },
+      { subject: 'Test' },
+      { subject: 'Test2' }
+    ]
+
+    it('should filter by differect predicates', () => {
```
Comment (pdobrosi) [inline test/utils.test.js:25]: typo: different

**Review by pdobrosi — APPROVED** [review]
I also wanted to implement the same feature :) , I like your solution.

**Comment by soulteary** [comment]
This feature looks very useful, I agree with @pdobrosi .

But the current code seems to have some conflicts, I will perform functional verification and resolve conflicts in a private fork.

If everything works, your name will appear in the v2.0.2 version, as a contributor, please continue to submit and play together @madflow 

@kowalczyka webui need update, we can talk it at https://github.com/maildev/maildev/issues/381

**Pull request closed without merging**